mousestyles/path_diversity/clean_movements.py

```diff
@@ -0,0 +1,108 @@
+import pandas as pd
+
+def clean_movements(movements, keep_index=None):
+    r"""
+    Returns a list of cleaned movement objects.
+
+    For each element of the input `movements` object, detects the existence of
+    rows which have i) same x,y-coordinates, or ii) same timestamps.
+    For case i), and removes the duplicated rows except for the first.
+    For case ii), raises value error.
+
+    Parameters
+    ----------
+    movements : list
+        each element is pandas.DataFrame containing 
+        CT, CX, CY coordinates.
+        Should have length greater than 1.
+
+    kee_index : boolean
+        whether or not keep the original index.
+        Deafalt is False.
+        If false, cleaned movement object is re-indexed.
+
+    Returns
+    -------
+    cleaned-movements : list
+        each element is cleaned movement object.
+
+    Examples
+    --------
+    >>> strain_nums = [0,0,1,1,2,2]
+    >>> mouse_nums = [0,0,1,1,2,2]
+    >>> day_nums = [0,11,0,11,0,11]
+    >>> movements = [data.load_movement(strain_num, mouse_num, day_num) for \
+    >>> strain_num, mouse_num, day_num in zip(strain_nums, mouse_nums, day_nums)]
+    >>> cleaned_movements, num_same_locs = cleaning_movements(movements)
+    """
+    
+    if type(movements) is not list:
+        raise TypeError("movements must be a list")
+    
+    if len(movements) == 0:
+        raise ValueError("movements must contain at least 1 movement object")
+    
+    if keep_index == None:
+        keep_index = False
+
+    if type(keep_index) is not bool:
+        raise TypeError("keep_index must be bool")
+    
+    def testing_input_movement(one_movement):
+        # test for each element of `movements`
+        # 1) must be pandas DataFrame
+        if type(one_movement) is not pd.core.frame.DataFrame:
+            out = 1
+        # 2) must have columns named `x`, `y`, `t`
+        elif set(one_movement.keys()).issuperset(['x','y','t']) == False:
+            out = 2
+        # 3) must have length greater than 1
+        elif len(one_movement) <= 1:
+            out = 3
```

Comment:
This function is designed to clean a list of movement dataframes. Say one needed to clean a single movement df -- the function would return an error. What if (if it doesn't change the rest of the code ... I don't think it does) we were to remove the <= 1 condition and allow for lists of length 1 to be passed into the function to allow for a single cleaning?